velox/expression/Expr.cpp

```diff
@@ -613,7 +613,11 @@ bool Expr::checkGetSharedSubexprValues(
         true /*override*/);
 
     evalEncodings(*missingRows, context, sharedSubexprValues_);
+
+    // Clear the rows which failed to compute.
+    context.deselectErrors(*sharedSubexprRows_);
   }
+
   context.moveOrCopyResult(sharedSubexprValues_, rows, result);
```

Comment:
If we are deselecting rows that resulted in an error, then the next time when we try to re-use the results from this shared expression, would the context have those rows marked as errors in this other branch of the expression tree? Basically wondering if we need to keep track of which rows threw errors and pass that along with the rest of the cached result.
Say if the expression was wrapped in Try() in one branch and not in the other?

(P.S. I am not too familiar with how errors are propagated across the expression tree)